crates/utils/re_crash_handler/src/sigint.rs

```diff
@@ -0,0 +1,47 @@
+use std::sync::atomic::AtomicBool;
+
+// ---
+
+static SIGINT_RECEIVED: AtomicBool = AtomicBool::new(false);
+
+/// Call this to start tracking `SIGINT`s.
+///
+/// You can then call [`was_sigint_ever_caught`] at any point in time.
+#[cfg(not(any(target_os = "windows", target_arch = "wasm32")))]
+#[allow(unsafe_code)]
+#[allow(clippy::fn_to_numeric_cast_any)]
+pub fn track_sigint() {
+    static ONCE: std::sync::Once = std::sync::Once::new();
+
+    ONCE.call_once(|| {
+        // SAFETY: we're installing a signal handler.
+        unsafe {
+            libc::signal(
+                libc::SIGINT,
+                signal_handler as *const fn(libc::c_int) as libc::size_t,
+            );
+        }
+
+        unsafe extern "C" fn signal_handler(signum: libc::c_int) {
+            SIGINT_RECEIVED.store(true, std::sync::atomic::Ordering::Relaxed);
+
+            // SAFETY: we're calling a signal handler.
+            unsafe {
+                libc::signal(signum, libc::SIG_DFL);
+                libc::raise(signum);
+            }
+        }
+    });
+}
+
+#[cfg(any(target_os = "windows", target_arch = "wasm32"))]
+#[allow(unsafe_code)]
+#[allow(clippy::fn_to_numeric_cast_any)]
+pub fn track_sigint() {}
+
+/// Returns whether a `SIGINT` was ever caught.
+///
+/// Need to call [`track_sigint`] at least once first.
+pub fn was_sigint_ever_caught() -> bool {
```

Comment:
This could call `track_sigint`, making it impossible to forget to call it. And then `track_signint` could just be made private.

Reply:
It would be too late by then though, you'd always miss the first SIGINT at the very least.

It's still worth doing though, at least if someone forget, SIGINT-spam will still work.